federation/src/main/scala/caliban/federation/package.scala

```diff
@@ -1,3 +1,12 @@
 package caliban
 
-package object federation extends FederationV1 with FederationDirectives
+import caliban.federation.v2.{ FederationDirectivesV2_3, FederationV2, Versions }
+
+package object federation {
+
+  lazy val v1   = new FederationV1 with FederationDirectives
+  lazy val v2_0 = new FederationV2(List(Versions.v2_0))
+  lazy val v2_1 = new FederationV2(List(Versions.v2_1))
+  lazy val v2_3 = new FederationV2(List(Versions.v2_3)) with FederationDirectivesV2_3
```

Comment:
I think it might be good to add a comment here on which one to use to retain the same behaviour as the old `v2` package. It might be unclear to users which one is the drop-in replacement. We might want to add that to the release notes / docs as well.

Reply:
See below, I was able to keep the `v2` constant but mark it deprecated to move users onto the new package structure